packages/openactive-broker-microservice/app.js

```diff
@@ -111,30 +127,42 @@ function getRPDE(url, contextIdentifier, cb) {
         } else if (VERBOSE) log(`Sleep mode poll for RPDE feed "${url}"`);
         context.sleepMode = true;
         if (context.timeToHarvestCompletion === undefined) context.timeToHarvestCompletion = millisToMinutesAndSeconds(new Date() - startTime);
-        setTimeout(() => getRPDE(url, contextIdentifier, cb), 500);
+        await sleep(500);
       } else {
-        context.responseTimes.push(response.elapsedTime);
+        context.responseTimes.push(responseTime);
         // Maintain a buffer of the last 5 items
         if (context.responseTimes.length > 5) context.responseTimes.shift();
         context.pages += 1;
         context.items += json.items.length;
         delete context.sleepMode;
-        cb(json, contextIdentifier);
+        if (REQUEST_LOGGING_ENABLED) {
+          const kind = json.items && json.items[0] && json.items[0].kind;
+          log(
+            `RPDE kind: ${kind}, page: ${context.pages}, length: ${
+              json.items.length
+            }, next: '${json.next}'`,
+          );
+        }
+        processPage(json, contextIdentifier);
+        url = json.next;
+      }
+    } catch (error) {
+      if (!error.response) {
+        log(`Error for RPDE feed "${url}": ${error.message}.`);
+        // Force retry, after a delay
+        await sleep(5000);
+      } else if (error.response.status === 404) {
+        if (WAIT_FOR_HARVEST) feedUpToDate(url);
+        log(`Not Found error for RPDE feed "${url}", feed will be ignored.`);
+        // Stop polling feed
+        return;
+      } else {
+        log(`Error ${error.response.status} for RPDE page "${url}": ${error.message}. Response: ${typeof error.response.data === 'object' ? JSON.stringify(error.response.data, null, 2) : error.response.data}`);
```

Comment:
Still annoying that it could be either (I assume? It wasn't clear from the docs...), and that you have to check to see which, but perhaps I've spent too much time in C#